### DISCUSSION TITLE
Squash Merge and Delete [Branch]

### DISCUSSION THREAD

**Discussion by MrSwitch** [community/community#120069] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

We have problems when developers Squash Merge when they should just Merge all the commits and vice-versa.

A Squash merge makes the source branch commits redundant in the project and I believe is always followed by deleting the branch. So why not just make them one thing. 

**Squash Merge and Delete [Branch]**

Providing this as an option in place of Squash Merge would be great because.
- It forces users to think about the action they are trying to do. 
- Sync PR's between protected branches would Ideally not even show this as an option, because they are undeletable. This would be great and address another ticket in these Discussions [Disable "Squash and merge" on some specific branches](https://github.com/orgs/community/discussions/10809).

**Comment by muspriandi** [comment]
Hi 👋
That’s a really good suggestion, many teams run into this same problem where people use Squash Merge or Merge Commit by mistake.

You’re right: a Squash Merge usually makes the old branch unnecessary because all commits are combined into one, and most users delete the branch right after.

Having a single option like “Squash Merge and Delete Branch” would make the workflow cleaner and help people think about what they’re doing before pressing merge.
